src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SslOverTdsStream.cs

```diff
@@ -160,7 +161,12 @@ public override void SetLength(long value)
         /// </summary>
         public override void Flush()
         {
-            _stream.Flush();
+            // Can sometimes get Pipe broken errors from flushing a PipeStream.
+            // PipeStream.Flush() also doesn't do anything, anyway.
+            if (!(_stream is PipeStream))
```

Comment:
Is it ever possible for _stream to be some intermediary wrapper around a PipeStream, like a BufferedStream?  I'm wondering if instead of trying to special-case it, it'd be better to just eat any IOException that emerges from the _stream.Flush call?